eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingServiceComposite.java

```diff
@@ -603,11 +602,25 @@ protected Object calculate() {
                 if (unwrapped != null || copyNamespaces != null) {
                     final MessageComposerType msgComposer = SOAPFactory.eINSTANCE
                             .createMessageComposerType();
-                    if (unwrapped != null) {
+                    final MessageComposerType oldMsgComposer = _binding.getMessageComposer();
+                    boolean unwrappedChanged = true;
```

Comment:
What do you think to extract a method isUnWrappedChanges(oldMsgComposer , unwrapped)  ?